app/views/catalog/_part_header_default.html.erb

```diff
@@ -6,7 +6,17 @@
     <% if document['ms_author_date_tesim'].present? %>
       <span class="part-author-date"><%= show_presenter.field_value('ms_author_date_tesim') %>.</span>
     <% end %>
-    <span class="part-supplied-title"><%= show_presenter.field_value('ms_supplied_title_tesim') %></span>
+
+    <span class="part-title">
+      <% if document['ms_title_tesim'] %>
+        <%= show_presenter.field_value('ms_title_tesim') %>
+      <% elsif document['ms_supplied_title_tesim'] %>
+        <%= show_presenter.field_value('ms_supplied_title_tesim') %>
+      <% elsif document['ms_uniform_title_tesim'] %>
+        <%= show_presenter.field_value('ms_uniform_title_tesim') %>
+      <% end %>
```

Comment:
Not really sure how we want to test this.  Our fixture data that we use for features doesn't really include this data (and doesn't appear to be fully hand-created).  I could push this into a helper and test that (but then we're getting into stubbing presenters, and its seems like a bit of a PITA).  Open to suggestions.

Reply:
The logic there seems pretty straight forward and readable. Maybe leave it for now, and see if we run into this down the road?